freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/flow/FreeIpaOrchestrationConfigService.java

```diff
@@ -43,45 +35,29 @@
     private StackService stackService;
 
     @Inject
-    private FreeIpaConfigService freeIpaConfigService;
-
-    @Inject
-    private ProxyConfigService proxyConfigService;
+    private FreeIpaNodeUtilService freeIpaNodeUtilService;
 
     @Inject
-    private TagConfigService tagConfigService;
+    private SaltConfigProvider saltConfigProvider;
 
     @Inject
-    private TelemetryConfigService telemetryConfigService;
-
-    @Inject
-    private FreeIpaNodeUtilService freeIpaNodeUtilService;
+    private CcmParametersConfigService ccmParametersConfigService;
 
     public void configureOrchestrator(Long stackId) throws CloudbreakOrchestratorException {
         Stack stack = stackService.getByIdWithListsInTransaction(stackId);
         Set<InstanceMetaData> instanceMetaDatas = stack.getNotDeletedInstanceMetaDataSet();
         List<GatewayConfig> gatewayConfigs = gatewayConfigService.getGatewayConfigs(stack, instanceMetaDatas);
         Set<Node> allNodes = freeIpaNodeUtilService.mapInstancesToNodes(instanceMetaDatas);
-
-        createAndPushPillarsToNodes(stackId, stack, gatewayConfigs, allNodes);
+        SaltConfig saltConfig = saltConfigProvider.getSaltConfig(stack, allNodes);
+        decoratePillars(stack.getAccountId(), saltConfig, stack.getCcmParameters());
+        hostOrchestrator.initSaltConfig(stack, gatewayConfigs, allNodes, saltConfig, new StackBasedExitCriteriaModel(stackId));
     }
 
-    @VisibleForTesting
-    SaltConfig getSaltConfig(Stack stack, Set<Node> hosts) throws CloudbreakOrchestratorFailedException {
-        SaltConfig saltConfig = new SaltConfig();
-        Map<String, SaltPillarProperties> servicePillarConfig = saltConfig.getServicePillarConfig();
-        FreeIpaConfigView freeIpaConfigView = freeIpaConfigService.createFreeIpaConfigs(stack, hosts);
-        servicePillarConfig.put("freeipa", new SaltPillarProperties("/freeipa/init.sls", Collections.singletonMap("freeipa", freeIpaConfigView.toMap())));
-        servicePillarConfig.put("discovery", new SaltPillarProperties("/discovery/init.sls", singletonMap("platform", stack.getCloudPlatform())));
-        servicePillarConfig.putAll(telemetryConfigService.createTelemetryPillarConfig(stack));
-        servicePillarConfig.putAll(proxyConfigService.createProxyPillarConfig(stack.getEnvironmentCrn()));
-        servicePillarConfig.putAll(tagConfigService.createTagsPillarConfig(stack));
-        return saltConfig;
+    private void decoratePillars(String accountId, SaltConfig saltConfig, CcmConnectivityParameters ccmParameters) {
+        if (ccmParameters != null && ccmParameters.getCcmV2JumpgateParameters() != null) {
+            Map<String, SaltPillarProperties> pillars = saltConfig.getServicePillarConfig();
+            pillars.putAll(ccmParametersConfigService.createCcmParametersPillarConfig(accountId, ccmParameters.getCcmV2JumpgateParameters()));
```

Comment:
instead of adding it here, you should return with a map
when the if is false, you could return with an empty map and use `putAll` in `configureOrchestrator`
also please rename the method.

or even better, you could have this logic in `SaltConfigProvider` where the other pillars are created

Reply:
ok